yb-voyager/src/srcdb/postgres.go

```diff
@@ -1153,9 +1153,62 @@ func (pg *PostgreSQL) GetMissingAssessMigrationPermissions() ([]string, error) {
 		combinedResult = append(combinedResult, fmt.Sprintf("\n%s[%s]", color.RedString("Missing SELECT permission for user %s on Tables: ", pg.source.User), strings.Join(missingTables, ", ")))
 	}
 
+	result, err := pg.checkPgssInstalledProperlyWithReadPermission()
+	if err != nil {
+		return nil, fmt.Errorf("error checking pg_stat_statement extension installed with read permissions: %w", err)
+	}
+
+	if len(result) > 0 {
+		combinedResult = append(combinedResult, result)
+	}
 	return combinedResult, nil
 }
 
+func (pg *PostgreSQL) checkPgssInstalledProperlyWithReadPermission() (string, error) {
+	// 1. check if pg_stat_statements extension is available on source
+	query := `SELECT nspname FROM pg_extension e, pg_namespace n
+	WHERE e.extnamespace = n.oid AND e.extname = 'pg_stat_statements'`
+	var pgssExtSchema string
+	err := pg.db.QueryRow(query).Scan(&pgssExtSchema)
+	if err != nil && err != sql.ErrNoRows {
+		return "", fmt.Errorf("failed to fetch the schema of pg_stat_statement available in: %w", err)
+	}
+	if pgssExtSchema == "" {
+		return "pg_stat_statements extension is not installed on source DB", nil
+	} else {
+		// check schema list
+		schemaList := lo.Union(pg.getTrimmedSchemaList(), []string{"public"})
+		if !slices.Contains(schemaList, pgssExtSchema) {
+			return fmt.Sprintf("pg_stat_statements extension schema %q is not in the schema list (%s)",
```

Comment:
Do we want this as guardrail as I remember what we discussed what is we will just check and if not found we continue?

Reply:
okay makes sense, 
add context for users  - `pg_stat_statements extension schema %q is not in the schema list (%s), required for detecting Unsupported Query Constructs`